src/utils/miscellaneous.ts

```diff
@@ -10,5 +10,13 @@ export function getToken<T extends TokenDetails>(
   value: string | undefined = '',
   tokens: T[] | undefined | null,
 ): T | undefined {
-  return (tokens || []).find(token => token[key] === value.toUpperCase())
+  const valueUppercase = value.toUpperCase()
+  return (tokens || []).find(token => {
+    const value = token[key]
+    if (value) {
+      return value.toString().toUpperCase() === valueUppercase
+    } else {
+      return false
+    }
```

Comment:
What about:

```suggest
return (tokens || []).find(token => (token[key] || '').toString().toUpperCase() === value.toUpperCase())
```

Too much clutter?

Reply:
I find to difficult to read these lines

Also, not that it matters much but you are converting to uppercase in each iteration of the token the `value` when it doesn't change